### PR TITLE
feat(gomod): do not shorten depName

### LINKED ISSUE #8691: Do not shortening go modules
**What Renovate type, platform and version are you using?**
all

**Describe the bug**
Since a regression of #8076 renovate will shorten `depName`, see #8683 and #8075

**Relevant debug logs**

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [x] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Moved the depName shorten to the `depNameShort` compatibillity part.

Declaring it as feature to have more visibillity of the change.

## Context:

fixes #8691

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
@rarkins What do you think about the depNameShort helper?

File: lib/workers/repository/updates/flatten.ts [hunk 1]
```diff
@@ -13,6 +13,26 @@ import { generateBranchName } from './branch-name';
...
+ * Build depNameShort
+ * #8075, #8691
+ */
+function shortenDepName(depName: string): string | undefined {
+  if (!depName) {
+    return undefined;
+  }
+
+  if (depName.startsWith('gopkg.in/')) {
+    const [pkg] = depName.replace('gopkg.in/', '').split('.');
```
Comment (rarkins) [inline lib/workers/repository/updates/flatten.ts:26]: This is 100% applicable to Go Modules only, so not good to be in lib/workers

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/updates/flatten.ts (hunk 1 above)
Comment (viceice) [reply lib/workers/repository/updates/flatten.ts:26]: ok, so removing the compabillity shortening. I don't think we really need that

**Comment by rarkins** [comment]
This is going to cause all Go branchNames to change. Did they change previously when we removed depNameShort or were we using depNameShort in the branch name before?

**Comment by viceice** [comment]
> This is going to cause all Go branchNames to change. Did they change previously when we removed depNameShort or were we using depNameShort in the branch name before?

yes, they all changed before from long to short now, so this is a kind of revert

**Pull request merged**